### PR TITLE
Add support for injections for any layer type

### LINKED ISSUE #44: Support injections into any layer type (Part 1)
- Adding default layer `conv` into `init`

### PR DESCRIPTION
**Pull request by n3a9**
- Extends support for injections beyond just Conv2d. User can specify layer types as list for injections.
- Default injection layers is still Conv2d, to support legacy code.
- Abstract hook additions into PyTorch model

Closes #44

### CODE REVIEW

File: pytorchfi/core.py [hunk 1]
```diff
@@ -66,6 +67,28 @@ def __init__(self, model, h, w, batch_size, **kwargs):
...
         )
 
+    def _traverseModelAndSetHooks(self, model, layer_types):
+        handles = []
+        shape = []
+        for layer in model.children():
+            # leaf node
+            if list(layer.children()) == []:
+                for i in layer_types:
+                    if isinstance(layer, i):
```
Comment (n3a9) [inline pytorchfi/core.py:77]: Will this append the hook to every instance of the provided layer? Or is there an option where we're able to just inject into a specific layer (for example, the first)

Comment (ma3mool) [reply pytorchfi/core.py:77]: At this stage, it appends the hook to all instances of `layer_type[i]`

During injection, the user can specify which specific layer of interest to perform the injection.

tl;dr: the hook is appended to all the layers. The injection location is specified later on by the user.

**Review by ma3mool — APPROVED** [review]
